internal/webhooks/runtime/extensionconfig_webhook_test.go

```diff
@@ -100,3 +104,198 @@ func TestExtensionConfigValidationFeatureGated(t *testing.T) {
+
+	extensionWithNoServiceNamespace := extensionWithService.DeepCopy()
+	extensionWithNoServiceNamespace.Spec.ClientConfig.Service.Namespace = ""
+
+	badURLExtension := extensionWithURL.DeepCopy()
+	badURLExtension.Spec.ClientConfig.URL = pointer.String("https//extension-address.com")
+
+	badSchemeExtension := extensionWithURL.DeepCopy()
+	badSchemeExtension.Spec.ClientConfig.URL = pointer.String("unknown://extension-address.com")
+
+	extensionWithInvalidServicePort := extensionWithService.DeepCopy()
+	extensionWithInvalidServicePort.Spec.ClientConfig.Service.Port = pointer.Int32(90000)
+
+	tests := []struct {
+		name        string
+		in          *runtimev1.ExtensionConfig
+		old         *runtimev1.ExtensionConfig
+		featureGate bool
+		expectErr   bool
+	}{
+		{
+			name:        "creation should fail if feature flag is disabled",
+			in:          extensionWithURL,
+			featureGate: false,
+			expectErr:   true,
+		},
+		{
+			name:        "update should fail if feature flag is disabled",
+			old:         extensionWithURL,
+			in:          updatedExtension,
+			featureGate: false,
+			expectErr:   true,
+		},
+		{
+			name:        "creation should fail if no name is defined",
+			in:          extensionWithNoServiceName,
+			featureGate: true,
+			expectErr:   true,
+		},
+		{
+			name:        "creation should fail if name violates Kubernetes naming rules",
+			in:          extensionWithBadServiceName,
+			featureGate: true,
+			expectErr:   true,
+		},
+		{
+			name:        "creation should fail if no namespace is defined",
+			in:          extensionWithNoServiceNamespace,
+			featureGate: true,
+			expectErr:   true,
+		},
+		{
+			name:        "update should fail if URL is invalid",
+			old:         extensionWithURL,
+			in:          badURLExtension,
+			featureGate: true,
+			expectErr:   true,
+		},
+		{
+			name:        "update should fail if URL scheme is invalid",
+			old:         extensionWithURL,
+			in:          badSchemeExtension,
+			featureGate: true,
+			expectErr:   true,
+		},
+		{
+			name:        "update should fail if URL and Service are both nil",
+			old:         extensionWithURL,
+			in:          extensionWithoutURLOrService,
+			featureGate: true,
+			expectErr:   true,
+		},
+		{
+			name:        "update should fail if both URL and Service are defined",
+			old:         extensionWithService,
+			in:          extensionWithServiceAndURL,
+			featureGate: true,
+			expectErr:   true,
+		},
+		{
+			name:        "update should fail if Service Path is invalid",
+			old:         extensionWithService,
+			in:          extensionWithInvalidServicePath,
+			featureGate: true,
+			expectErr:   true,
+		},
+
+		{
+			name:        "update should fail if Service Port is invalid",
+			old:         extensionWithService,
+			in:          extensionWithInvalidServicePort,
+			featureGate: true,
+			expectErr:   true,
+		},
+		{
+			name:        "update should pass if Service Path is valid",
```

Comment:
Out of curiosity: Is there any specific reason to add a pass test case only for valid path?

It doesn't look like the test case is looking for anything specific for the path. It might be good to have a passing test case, so simply rename the test case to something around `update should pass if updated extension is valid`. Thoughts?

Reply:
Makes sense 👍  did change the name.